### ISSUE TITLE
kubernetes service can't be ready

### ISSUE THREAD

**Issue by dfang** [hashicorp/waypoint#584]
**Describe the bug**

waypoint service can't be ready

```
$ waypoint install --platform=kubernetes -accept-tos
service/waypoint created
statefulset.apps/waypoint-server created
⠏ Waiting for Kubernetes service to be ready...

```
**Steps to Reproduce**
```
Client: Docker Engine - Community
 Cloud integration  0.1.18
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

$ k get svc 
waypoint         LoadBalancer   10.105.50.118    <pending>     9701:31084/TCP,9702:30375/TCP   90s

```

i'm using docker for mac edge version, it supports load balancer type, i checked it with traefik helm chart

```
helm install traefik traefik/traefik
k get svc | grep traefik
```

**Expected behavior**

```
the waypoint svc is ready
```

**Comment by mitchellh** [comment]
I’m unable to reproduce this unfortunately... any other details? If we can’t reproduce this or hear from anyone else with this issue we won’t be able to resolve this. 😭

**Comment by dfang** [comment]
Hi mitchellh, 

i tried docker tutorial before kubernetes tutorial, there is a waypoint docker container, so i removed it, then run `waypoint install --platform=kubernetes -accept-tos`. but i got another issue:

```
/private/tmp λ waypoint install --platform=kubernetes -accept-tos
service/waypoint created
statefulset.apps/waypoint-server created
! Error getting the initial token: server is already bootstrapped

  The Waypoint server has been deployed, but due to this error we were
  unable to automatically configure the local CLI or the Waypoint server
  advertise address. You must do this manually using "waypoint context"
  and "waypoint server config-set".
```

**Comment by naomichi-y** [comment]
I'm getting a similar error... :joy:

```
% kind create cluster --name=sandbox
Creating cluster "sandbox" ...
 ✓ Ensuring node image (kindest/node:v1.19.1) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-sandbox"
You can now use your cluster with:

kubectl cluster-info --context kind-sandbox

Have a nice day! 👋

% cd waypoint-examples/kubernetes/nodejs
% waypoint init
✓ Configuration file appears valid
✓ Local mode initialized successfully
✓ Project "example-nodejs" and all apps are registered with the server.
✓ Plugins loaded and configured successfully
✓ Authentication requirements appear satisfied.

Project initialized!

You may now call 'waypoint up' to deploy your project or
commands such as 'waypoint build' to perform steps individually.

% waypoint init
✓ Configuration file appears valid
✓ Local mode initialized successfully
✓ Project "example-nodejs" and all apps are registered with the server.
✓ Plugins loaded and configured successfully
✓ Authentication requirements appear satisfied.

Project initialized!

You may now call 'waypoint up' to deploy your project or
commands such as 'waypoint build' to perform steps individually.

% waypoint up

» Building...
Creating new buildpack-based image using builder: heroku/buildpacks:18
✓ Creating pack client
✓ Building image
 │ [exporter] Adding 1/1 app layer(s)
 │ [exporter] Adding layer 'launcher'
 │ [exporter] Adding layer 'config'
 │ [exporter] Adding label 'io.buildpacks.lifecycle.metadata'
 │ [exporter] Adding label 'io.buildpacks.build.metadata'
 │ [exporter] Adding label 'io.buildpacks.project.metadata'
 │ [exporter] *** Images (deac2c0d070b):
 │ [exporter]       index.docker.io/library/example-nodejs:latest
 │ [exporter] Adding cache layer 'heroku/nodejs-engine:nodejs'
 │ [exporter] Adding cache layer 'heroku/nodejs-engine:toolbox'
✓ Injecting entrypoint binary to image

Generated new Docker image: example-nodejs:latest
✓ Tagging Docker image: example-nodejs:latest => nodejs-example:latest

» Deploying...
✓ Kubernetes client connected to https://127.0.0.1:57257 with namespace default
✓ Creating deployment...
⠋ Waiting on deployment to become available: 1/0/0
```

Docker log contained some errors.
```
% docker ps
doCONTAINER ID        IMAGE                  COMMAND                  CREATED             STATUS              PORTS                       NAMES
2c1f49db0ef7        kindest/node:v1.19.1   "/usr/local/bin/entr…"   5 minutes ago       Up 5 minutes        127.0.0.1:57257->6443/tcp   sandbox-control-plane
[naomichi@Naomichis-MacBook-Pro] ~/Repositories/waypoint-examples/kubernetes/nodejs
% docker logs 2c1f49db0ef7
INFO: ensuring we can execute mount/umount even with userns-remap
INFO: remounting /sys read-only
INFO: making mounts shared
INFO: fix cgroup mounts for all subsystems
INFO: clearing and regenerating /etc/machine-id
Initializing machine ID from random generator.
INFO: faking /sys/class/dmi/id/product_name to be "kind"
INFO: faking /sys/class/dmi/id/product_uuid to be random
INFO: faking /sys/devices/virtual/dmi/id/product_uuid as well
INFO: setting iptables to detected mode: legacy
INFO: Detected IPv4 address: 172.18.0.2
INFO: Detected IPv6 address: ::ffff:172.18.0.2
Failed to find module 'autofs4'
systemd 246.2-1ubuntu1 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid)
Detected virtualization docker.
Detected architecture x86-64.
Failed to create symlink /sys/fs/cgroup/net_prio: File exists
Failed to create symlink /sys/fs/cgroup/net_cls: File exists
Failed to create symlink /sys/fs/cgroup/cpuacct: File exists
Failed to create symlink /sys/fs/cgroup/cpu: File exists

Welcome to Ubuntu Groovy Gorilla (development branch)!

Set hostname to <sandbox-control-plane>.
Queued start job for default target Graphical Interface.
[  OK  ] Started Dispatch Password …ts to Console Directory Watch.
[UNSUPP] Starting of Arbitrary Exec…Automount Point not supported.
[  OK  ] Reached target Local Encrypted Volumes.
[  OK  ] Reached target Paths.
[  OK  ] Reached target Slices.
[  OK  ] Reached target Swap.
[  OK  ] Listening on Journal Audit Socket.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Listening on Journal Socket.
[  OK  ] Reached target Sockets.
         Mounting Huge Pages File System...
         Mounting Kernel Debug File System...
         Mounting Kernel Trace File System...
         Starting Journal Service...
         Starting Create list of st…odes for the current kernel...
         Mounting FUSE Control File System...
         Starting Remount Root and Kernel File Systems...
         Starting Apply Kernel Variables...
[  OK  ] Mounted Huge Pages File System.
[  OK  ] Mounted Kernel Debug File System.
[  OK  ] Mounted Kernel Trace File System.
[  OK  ] Finished Create list of st… nodes for the current kernel.
[  OK  ] Mounted FUSE Control File System.
[  OK  ] Finished Remount Root and Kernel File Systems.
         Starting Create System Users...
         Starting Update UTMP about System Boot/Shutdown...
[  OK  ] Finished Apply Kernel Variables.
[  OK  ] Finished Update UTMP about System Boot/Shutdown.
[  OK  ] Finished Create System Users.
         Starting Create Static Device Nodes in /dev...
[  OK  ] Finished Create Static Device Nodes in /dev.
[  OK  ] Reached target Local File Systems (Pre).
[  OK  ] Reached target Local File Systems.
[  OK  ] Started Journal Service.
[  OK  ] Reached target System Initialization.
[  OK  ] Started Daily Cleanup of Temporary Directories.
[  OK  ] Reached target Basic System.
[  OK  ] Reached target Timers.
         Starting containerd container runtime...
[  OK  ] Started kubelet: The Kubernetes Node Agent.
         Starting Flush Journal to Persistent Storage...
[  OK  ] Started containerd container runtime.
[  OK  ] Reached target Multi-User System.
[  OK  ] Reached target Graphical Interface.
         Starting Update UTMP about System Runlevel Changes...
[  OK  ] Finished Flush Journal to Persistent Storage.
[  OK  ] Finished Update UTMP about System Runlevel Changes
```

I'm using Docker for Mac and k8s settings are as follows:

<img width="1007" alt="Screen Shot 2020-10-19 at 2 23 22" src="https://user-images.githubusercontent.com/1632478/96375131-248cf300-11b2-11eb-8412-437a1d98b703.png">

**Comment by mitchellh** [comment]
@dfang This is normal see https://www.waypointproject.io/docs/troubleshooting

You have to fully remove your existing server before reinstalling. Based on that output though I'm confident your next try will work since that is farther than the first issue reported. 

@naomichi-y I think you're hitting `kind` issues, we have a number of open issues around kind we're going to resolve.

**Comment by ronakbanka** [comment]
@naomichi-y check this to use k8s cluster from Docker Desktop, works like a charm

https://medium.com/@ronakbanka/getting-started-with-hashicorp-waypoint-and-kubernetes-c818607c9b05
